brave/brave-core
# Add Rewards settings to brave://settings

Resolves https://github.com/brave/brave-browser/issues/18158

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Screenshot:
![rewards-settings](https://user-images.githubusercontent.com/5033691/138104169-73270a8a-b851-4e29-9a21-76b48bf5d623.png)

## Test Plan:

- Clean profile
- Visit brave://settings
- Confirm that Rewards section is empty except for message about enabling Rewards
- Click the link in the above message
- Verify that Rewards panel opens
- Enable Rewards
- Verify that brave://settings page updates with Rewards settings and that settings reflect current Rewards state
- Verify that changing settings in brave://settings reflects the same changes in brave://rewards and vice versa
- Verify that enabling/disabling main options enables/disables sub-options
- Verify that preference changes persist across browser restarts
- Verify that dropdowns with "Default" selected work as expected (i.e., use the default value for that preference)
- Verify that Ads state dropdown works as expected, including Auto and Disabled settings
